src/interfaces/chain.cpp

```diff
@@ -358,7 +344,8 @@ class ChainImpl : public Chain
     {
         if (!old_tip.IsNull()) {
             LOCK(::cs_main);
-            if (old_tip == ::ChainActive().Tip()->GetBlockHash()) return;
+            const CChain& active = m_node.chainman->ActiveChain();
```

Comment:
```suggestion
            const CChain& active = Assert(m_node.chainman)->ActiveChain();
```

style-nit, no strong opinion, though.